backend/src/auth/services/auth_service.py

```diff
@@ -1,15 +1,67 @@
+import uuid
+from datetime import timedelta
+from typing import Optional
+
+from fastapi import HTTPException
+from fastapi.responses import JSONResponse
 from sqlalchemy.orm import Session
 
-from src.auth.interfaces.repository import Repository
-from src.auth.schemas import SignInSchema, UserSchema
+from src.auth.interfaces import Repository
+from src.auth.schemas import (
+    RegisterRequest,
+    RegisterResponse,
+    SignInRequest,
+)
+from src.auth.utils import return_token
 
 
 class AuthService:
     def __init__(self, repository: Repository):
         self._repository = repository
 
-    def register_user(self, user: UserSchema, db: Session):
-        return self._repository.register(user, db)
+    @staticmethod
+    def generate_token_response(
+        username: str, user_id: uuid.UUID
+    ) -> Optional[JSONResponse]:
+        try:
+            token = return_token(username, user_id)
+            refresh_token = token.pop("refresh_token")
+            response = JSONResponse(content=token)
+            response.set_cookie(
+                key="refresh_token",
+                value=refresh_token,
+                httponly=True,
+                expires=int(timedelta(days=365).total_seconds()),
+            )
+            return response
```

Comment:
I dont understand why are you setting cookie instead of just return normal token ?

Reply:
soo never return cookie or header? jsons only?

Reply:
Exactly